tooling/ssa_fuzzer/fuzzer/fuzz_targets/field.rs

```diff
@@ -0,0 +1,473 @@
+                self.insert_array_get(array, index);
+            }
+            Instructions::ArraySet { array, index, value } => {
+                self.insert_array_set(array, index, value);
+            }
+            Instructions::BigCastAndBack { lhs, size } => {
+                let acir_len = self.acir_ids.len() as u32;
+                let brillig_len = self.brillig_ids.len() as u32;
+                let acir_lhs = u32_to_id_value(self.acir_ids[(lhs % acir_len) as usize]);
+                let brillig_lhs = u32_to_id_value(self.brillig_ids[(lhs % brillig_len) as usize]);
+                let acir_result = self.acir_builder.insert_cast_bigger_and_back(acir_lhs, size);
+                let brillig_result =
+                    self.brillig_builder.insert_cast_bigger_and_back(brillig_lhs, size);
+                self.acir_ids.push(id_to_int(acir_result));
+                self.brillig_ids.push(id_to_int(brillig_result));
+            }
+            _ => {}
+        }
+    }
+
+    /// Finalizes the function by setting the return value
+    fn finalize_function(&mut self) {
+        let acir_result_index = *self.acir_ids.last().unwrap();
+        let brillig_result_index = *self.brillig_ids.last().unwrap();
+        self.acir_builder.finalize_function(u32_to_id_value(acir_result_index));
+        self.brillig_builder.finalize_function(u32_to_id_value(brillig_result_index));
+    }
+
+    /// Returns witnesses for ACIR and Brillig
+    /// Only one witness added as return value is the last variable
+    /// If acir_witnesses_indeces or brillig_witnesses_indeces are not equal to NUMBER_OF_VARIABLES_INITIAL,
+    /// then it means that no instructions were added, so we just return the last variable set
+    fn get_return_witnesses(&mut self) -> (Witness, Witness) {
+        let acir_result_index = *self.acir_ids.last().unwrap();
+        let brillig_result_index = *self.brillig_ids.last().unwrap();
+        let mut acir_result_witness = Witness(acir_result_index);
+        let mut brillig_result_witness = Witness(brillig_result_index);
+
+        if self.acir_ids.len() as u32 != config::NUMBER_OF_VARIABLES_INITIAL {
+            acir_result_witness = Witness(NUMBER_OF_VARIABLES_INITIAL);
+            brillig_result_witness = Witness(NUMBER_OF_VARIABLES_INITIAL);
+        }
+        (acir_result_witness, brillig_result_witness)
+    }
+
+    /// Returns programs for ACIR and Brillig
+    fn get_programs(
+        self,
+    ) -> (Result<CompiledProgram, CompileError>, Result<CompiledProgram, CompileError>) {
+        (self.acir_builder.compile(), self.brillig_builder.compile())
+    }
+}
+
+/// Represents the data for the fuzzer
+/// `methods` - sequence of instructions to be added to the program
+/// `initial_witness` - initial witness values for the program as String
+#[derive(Arbitrary, Debug, Clone, Hash)]
+struct FuzzerData {
+    methods: Vec<Instructions>,
+    initial_witness: [String; config::NUMBER_OF_VARIABLES_INITIAL as usize],
```

Comment:
Can this be any string whatsoever? Can't you create some new type and implement `Arbitrary` for it so that it generates valid fields?